src/main/java/io/openraven/magpie/api/cspm/Policy.java

```diff
@@ -0,0 +1,98 @@
+/*-
+ * #%L
+ * Magpie API
+ * %%
+ * Copyright (C) 2021 Open Raven Inc
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package io.openraven.magpie.api.cspm;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Policy {
```

Comment:
Let's talk on standup tomorrow, I'm a little confused by us putting all the Policy/Violation objects into `magpie-api`, I thought we were going to make it generic and keep these objects in `magpie-core`?

Reply:
yes, last time I purely got your point.

But due to that, we are getting into trouble were each party provide its own implementation of ScanResult.
- magpie-core operates with its own class ScanResult which will be filled out with data
- magpie-policy-output create its own ScanResult to define how the data will be retrieved from it and rendered/processed
We dont want to wire dependency of one onto another think so coz in any case it is not flexible (core to output or vise versa) 
Since the cspm-core invoke and delegate ScanResult object to policy-output plugin, we are lacking here the shared contract between them
This way we fail with into class cast problems in runtime -> when plugin is loaded and Scan result prepared, we just cannot pass it coz pluggin depend on ScanResult provided on its own packages even if it has the same definition.
So we need some third party dependency with will define the ScanResult class for both producer (cspm-core) and consumer (policy-output).

One possible solution here is to serialize/deserialize ScanResult while passing to the output-plugin. 
Thats will segregate producer from consumer API completely.
But we need to discuss this.

Reply:
Will create a branch tomorrow with arch problem about which Im talking above. So we will have an example